libstuff/SHTTPSProxySocket.cpp

```diff
@@ -70,7 +64,8 @@ bool SHTTPSProxySocket::send(const string& buffer, size_t* bytesSentCount) {
     if (state.load() < Socket::State::SHUTTINGDOWN) {
         if (!filledPreSendBuffer) {
             SData connectMessage("CONNECT " + hostname + " HTTP/1.1");
-            connectMessage["Host"] = proxyAddress;
+            connectMessage["Host"] = hostname;
```

Comment:
It turns out that this should be the destination hostname, not the proxy hostname.